### PR TITLE
[MIG][12.0] sale_report_margin: Migration to v12.0

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT20187
@chienandalu @rafaelbn Can you review?

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
Looks good :+1:

File: sale_report_margin/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Sale Report Margin",
+    "version": "12.0.1.0.0",
+    "author": "Tecnativa,"
+              "Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/sale-reporting",
+    "development_status": "Production/Stable",
```
Comment (pedrobaeza) [inline sale_report_margin/__manifest__.py:9]: You need to declare `maintainers` if promoting this to stable

File: sale_report_margin/reports/sale_report.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+class SaleReport(models.Model):
+    _inherit = "sale.report"
+
+    purchase_price = fields.Float(
+        string='Purchase Price',
+        readonly=True,
+    )
+
+    def _select(self):
```
Comment (pedrobaeza) [inline sale_report_margin/reports/sale_report.py]: This has been modified upstream. The hint about this is the coverage drop. Now everything goes in one method:

https://github.com/odoo/odoo/blob/a7d24cabafe1a9c71f9dbdb7cf3f129006250951/addons/sale/report/sale_report.py#L58

and in this case it's simple, as you only have to add the field in the argument:

```python
def _query(self, with_clause='', fields={}, groupby='', from_clause=''):
    fields['purchase_price'] = ',\nSUM(l.purchase_price / CASE COALESCE(s.currency_rate, 0)) as purchase_price'
    return super()._query(self, with_clause=with_clause, fields=fields, groupby=groupby, from_clause=from_clause)
```
My only doubt (but it's an Odoo problem) is if the argument `field={}` will have problem in a multi-DB (https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python).

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
Changes done!!

**Comment by pedrobaeza** [comment]
Everything is red

**Comment by sergio-teruel** [comment]
Changes done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**